server/controllers/user.controller.js

```diff
@@ -1,5 +1,86 @@
-export const test = (req, res) => {
-  res.json({
-    message: 'Hello it is api controller'
-  })
-}
\ No newline at end of file
+import User from "../models/user.model.js";
+import bcrypt from "bcrypt";
+
+
+// const fakeUser = new User({
+//   user_name: "Old Name",
+//   user_birth: new Date("1990-01-01"),
+//   user_gender: "Male",
+//   user_email: "oldemail@example.com",
+//   user_password: "oldpassword",
+//   user_phone: "1234567890",
+//   user_address: "Old Address",
+// });
+
+// await fakeUser.save();
+
+export const edit = async (req, res) => {
+  const {
+    user_name,
+    user_birth,
+    user_gender,
+    user_email,
+    user_phone,
+    user_address,
+  } = req.body;
+
+  try {
+    const user = await User.findByEmail(req.body.email);
+
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    user.user_name = user_name;
+    user.user_birth = user_birth;
+    user.user_gender = user_gender;
+    user.user_email = user_email;
+    user.user_phone = user_phone;
+    user.user_address = user_address;
+
+    await user.save();
+
+    res.json({
+      message: "User updated successfully",
+      user,
+    });
```

Comment:
Có thể sử dụng:

`
const checkUser = await User.findOneAndUpdate({ user_email }, req.body).exec();

if (!checkUser)
  return responseHandler.notFound(res, 'User not found');

return responseHandler.ok(res, 'User updated successfully');
`